src/index.js

```diff
@@ -64,8 +64,8 @@ class KasaBulbInstance extends InstanceBase {
 			}
 		}
 
-		this.getInformation()
-		this.setupInterval()
+		await this.getInformation()
```

Comment:
Awaiting this here could be problematic if the device is unreachable. If this takes longer than the timeout companion sets on the call to confogUpdated, then companion could kill and restart the connection.

Reply:
if the methods are async methods, shouldn't we await them?

Reply:
the promise the methods return needs to be handled, they dont necessarily need to be awaited.
```
this.getInformation().then(async () => {
 // do something here?
}).catch((e) => {
 // handle the error here
})
```
is a different way of handling the promise. the `then` is useful (but not necessary) to do something once the promise resolves. the `catch` is the important part, otherwise the uncaught error will cause the module's process to crash and be restarted.  
Which approach to use depends on how the execution should flow in the code

but as it stands, if that `await` takes longer than 5s, companion will see the module process as unresponsive and might restart it. 

In fact, the promise returned https://github.com/bitfocus/companion-module-tplink-kasasmartbulb/blob/master/src/index.js#L45 is not being handled. Which is lucky for now to avoid the module getting stuck in a bad state.